astropy/io/misc/asdf/tags/transform/projections.py

```diff
@@ -5,13 +5,21 @@
 
 from asdf import yamlutil
 
+import astropy.units as u
 from astropy import modeling
 from .basic import TransformType
 
 
 __all__ = ['AffineType', 'Rotate2DType', 'Rotate3DType']
 
 
+def _parameter_to_value(param):
+    if param.unit is not None:
+        return u.Quantity(param)
+    else:
+        return param.value
+
```

Comment:
Looks like this function could be useful in `polynomial.py` too. Would it be useful to refactor it into `transform/__init__.py`?